src/haddock/modules/analysis/ilrmsdmatrix/__init__.py

```diff
@@ -205,29 +214,58 @@ def _run(self):
+        lig_traj_filename = "traj_lig.xyz"
+        prev_keys = []
+        with open(rec_traj_filename, "w") as rec_traj_xyz:
+            with open(lig_traj_filename, "w") as lig_traj_xyz:
+                for mod in models:
+                    atoms = {}
+                    atoms.update(get_atoms(mod))
+                    ref_coord_dic, _ = load_coords(
+                    mod, atoms, res_resdic
+                    )
+                    if prev_keys != []:
+                        if ref_coord_dic.keys() != prev_keys:
+                            self.finish_with_error(
+                                "The keys of the ref_coord_dic are not the "
+                                "same for all the models. Please check the "
+                                "input models."
+                                )
+                    # write receptor coords
+                    rec_coords = {k:ref_coord_dic[k] for k in ref_coord_dic if k[0] == self.params["receptor_chain"]}
+                    lig_coords = {k:ref_coord_dic[k] for k in ref_coord_dic if k[0] in self.params["ligand_chains"]}
+                    n_rec_atoms = len(rec_coords)
+                    n_lig_atoms = len(lig_coords)
+                    rec_traj_xyz.write(f"{n_rec_atoms}{os.linesep}{os.linesep}")
+                    lig_traj_xyz.write(f"{n_lig_atoms}{os.linesep}{os.linesep}")
+                    for k, v in rec_coords.items():
+                        rec_traj_xyz.write(f"x {v[0]} {v[1]} {v[2]}{os.linesep}")
+                    for k, v in lig_coords.items():
+                        lig_traj_xyz.write(f"x {v[0]} {v[1]} {v[2]}{os.linesep}")
+                    
+                    prev_keys = ref_coord_dic.keys()
+        
+        # Calculate the rmsd for each set of models
+        ilrmsd_jobs: list[RMSDJob] = []
+        self.log(f"running RmsdFast Jobs with {ncores} cores")
         for core in range(ncores):
-            output_name = "ilrmsd_" + str(core) + ".matrix"
-            ilrmsd_obj = ilRMSD(
-                models,
-                core,
-                npairs[core],
-                ref_structs[core],
-                mod_structs[core],
-                res_resdic,
-                output_name,
-                path=Path("."),
-                params=self.params
-                )
+            output_name = Path("ilrmsd_" + str(core) + ".out")
             job_f = Path(output_name)
             # init RMSDJob
-            job = ilRMSDJob(
-                job_f,
-                self.params,
-                ilrmsd_obj
+            job = RMSDJob(
+                rec_traj_filename,
+                output_name,
+                rmsdmatrix_executable,
+                f"{core} ",
+                f"{npairs[core]}",
+                f"{ref_structs[core]}",
+                f"{mod_structs[core]}",
+                f"{len(models)}",
+                f"{n_rec_atoms}",
+                lig_traj_filename,
+                f"{n_lig_atoms}",
```

Comment:
With exception of `lig_traj_filename` it seems that you are passing numbers here as strings. Which only works because `BaseJob` takes `*args` - but do double check the correct types here and maybe use the appropriates ones here to avoid confusion.

Reply:
you're completely right about this, no need to pass the string at all